src/charm.py

```diff
@@ -52,7 +52,7 @@ def __init__(self, *args):
         self.framework.observe(self.on.update_status, self._on_update_status)
         self.framework.observe(self.on.upgrade_charm, self._on_upgrade_charm)
 
-    @lru_cache
+    @lru_cache()
     def _arch(self):
         architecture = check_output(["dpkg", "--print-architecture"]).rstrip()
         architecture = architecture.decode("utf-8")
```

Comment:
this looks like a good candidate to be a `cached_property`, but if it's not possible in this PR, this is a sane approach

```python
from functools import cached_property
...


    @cached_property
    def _arch(self):
        ...


```

but of course you'd have to change the usages to `charm._arch` rather than `charm._arch()`

Reply:
Cool, I'll change it to a `cached_property`. Thanks for the suggestion 😃